### PR TITLE
feat: validate x-sdk-operations extension

### PR DESCRIPTION
**Pull request by barrett-schonefeld**
Purpose:
- ensure the request examples provided using `x-sdk-operations` extension are good, so they may be utilized properly by tooling that uses `x-sdk-operations`

Changes:
- add custom Spectral rule to validate x-sdk-operations
- add custom Spectral function to invoke ajv JSON Schema validator on x-sdk-operations JSON Schema
- add JSON Schema representation of `x-sdk-operations`

Tests:
- ensure warnings issued for invalid x-sdk-operations schema

Docs:
- document `ibm-sdk-operations` rule

### CODE REVIEW

**Review by dpopp07 — APPROVED** [review]
Looks good to me! Left one comment on the README and we probably want to let Mike have a look at the schema before merging

File: docs/spectral-rules.md [hunk 1]
```diff
@@ -56,6 +56,35 @@ responses:
...
+
+```yaml
+responses:
+  200:
+    content:
+      application/json:
+        schema:
+          type: string
+```
+
```
[the remark this answers is not in the document]
Comment (barrett-schonefeld) [reply docs/spectral-rules.md]: Yes, thank you!

**Review by dpopp07 — APPROVED** [review]
Still looks good to me 👍

**Review by mkistler — APPROVED** [review]
Looks good! 👍

**Pull request merged**